pandas/tests/indexes/common.py

```diff
@@ -859,13 +863,11 @@ def test_getitem_2d_deprecated(self):
 
         assert isinstance(res, np.ndarray), type(res)
 
-    def test_contains_requires_hashable_raises(self):
+    @pytest.mark.parametrize("value", [[], {}])
+    def test_contains_requires_hashable_raises(self, value):
         idx = self.create_index()
         with pytest.raises(TypeError, match="unhashable type"):
-            [] in idx
-
-        with pytest.raises(TypeError):
-            {} in idx._engine
+            value in idx
```

Comment:
we also need to check for `value in idx._engine`

Reply:
Thanks @jbrockmendel  added it now.